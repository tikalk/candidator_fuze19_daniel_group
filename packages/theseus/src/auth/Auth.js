// import auth0 from 'auth0-js'; // no-unused-vars
import autoBind from 'react-autobind';
import history from './history';

export default class Auth {
  // auth0 = new auth0.WebAuth({
  //   domain: process.env.REACT_APP_AUTH0_DOMAIN,
  //   clientID: process.env.REACT_APP_AUTH0_CLIENT_ID,
  //   redirectUri: process.env.REACT_APP_AUTH0_REDIRECT_URI,
  //   audience: process.env.REACT_APP_AUTH0_AUDIENCE,
  //   scope: 'openid email profile read:questions',
  //   responseType: 'token id_token',
  // });

  static getAccessToken() {
    return localStorage.getItem('access_token');
  }

  constructor() {
    autoBind(this);
  }

  login() {
    // this.auth0.authorize();
    setTimeout(() => {
      this.setSession({});
      history.replace('/questions/');
    }, 0);
  }

  handleAuthentication() {
    this.auth0.parseHash((err, authResult) => {
      if (authResult && authResult.accessToken && authResult.idToken) {
        this.setSession(authResult);
        history.replace('/questions/');
      } else if (err) {
        history.replace('/');
        console.log(err); // eslint-disable-line
      }
    });
  }

  setSession(authResult) {
    // Set the time that the access token will expire at
    const expiresAt = JSON.stringify(100000 * 1000 + new Date().getTime());
    localStorage.setItem('access_token', '123');
    localStorage.setItem('id_token', '456');
    localStorage.setItem('expires_at', expiresAt);
    localStorage.setItem('email', 'aa@a.com');
    localStorage.setItem('sub', 'aa');
    localStorage.setItem('name', 'jon dir');
    localStorage.setItem('picture', '');
    localStorage.setItem('hkjsfhlav', authResult);

    // localStorage.setItem('access_token', authResult.accessToken);
    // localStorage.setItem('id_token', authResult.idToken);
    // localStorage.setItem('expires_at', expiresAt);
    // localStorage.setItem('email', authResult.idTokenPayload.email);
    // localStorage.setItem('sub', authResult.idTokenPayload.sub);
    // localStorage.setItem('name', authResult.idTokenPayload.nickname);
    // localStorage.setItem('picture', authResult.idTokenPayload.picture);

    // navigate to the home route
    history.replace('/questions/');
  }

  logout(redirect) {
    // Clear access token and ID token from local storage
    localStorage.removeItem('access_token');
    localStorage.removeItem('id_token');
    localStorage.removeItem('expires_at');
    localStorage.removeItem('email');
    localStorage.removeItem('sub');

    // navigate to the home route
    if (redirect) {
      history.replace('/');
    }
  }

  isAuthenticated() {
    // Check whether the current time is past the
    // access token's expiry time
    const expiresAt = JSON.parse(localStorage.getItem('expires_at'));
    return new Date().getTime() < expiresAt;
  }
}
