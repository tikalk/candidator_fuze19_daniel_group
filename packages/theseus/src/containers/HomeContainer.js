import React from 'react';
import { connect } from 'react-redux';
import { markdown } from 'markdown';
import Button from '../components/Button';
import Styled from './styled-components';
import text from './homepage-intro';

// const Options = ['Javascript', 'Java', 'Python'];

export function HomeContainer({ auth, language }) {
  const login = () => {
    auth.login(language);
  };
  return (
    <Styled.Content>
      <div dangerouslySetInnerHTML={{ __html: markdown.toHTML(text) }} />
      {!auth.isAuthenticated() && (
        <Styled.ButtonWrapper>
          {/* <Styled.Select onChange={v => setLanguage(v.target.value)}>
            {Options.map((option, i) => (
              <option key={`option-${i}`}>{option}</option>
            ))}
          </Styled.Select> */}
          <Button primary onClick={login}>
            Start
          </Button>
        </Styled.ButtonWrapper>
      )}
    </Styled.Content>
  );
}

function mapStateToProps(state) {
  return {
    language: state.candidator.language,
  };
}

// function mapDispatchToProps(dispatch) {
//   return {
//     setLanguage: bindActionCreators(CandidatorActions.setLanguage, dispatch),
//   };
// }

export default connect(mapStateToProps)(HomeContainer);
