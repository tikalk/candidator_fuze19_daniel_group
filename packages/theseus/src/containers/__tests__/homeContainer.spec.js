import React from 'react';
import renderer from 'react-test-renderer';
import { HomeContainer } from '../HomeContainer';
import Auth from '../../auth/Auth';

jest.mock('../../auth/Auth');

describe('Test <Header>', () => {
  beforeEach(() => {
    jest.resetModules();
  });

  it('renders correctly - not logged in', () => {
    const auth = new Auth();
    const tree = renderer.create(<HomeContainer auth={auth} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly - logged in', () => {
    const auth = new Auth();
    auth.isAuthenticated = jest.fn().mockImplementation(() => true);
    const tree = renderer.create(<HomeContainer auth={auth} history={{}} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
