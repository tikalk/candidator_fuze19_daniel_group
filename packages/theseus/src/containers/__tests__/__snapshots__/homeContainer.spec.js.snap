// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Header> renders correctly - logged in 1`] = `
<div
  className="sc-bdVaJa fKAxYV"
>
  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
</div>
`;

exports[`Test <Header> renders correctly - not logged in 1`] = `
<div
  className="sc-bdVaJa fKAxYV"
>
  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
  <div
    className="sc-htpNat dlKyGL"
  >
    <select
      className="sc-bxivhb VSJEm"
      onChange={[Function]}
    >
      <option>
        Javascript
      </option>
      <option>
        Java
      </option>
      <option>
        Python
      </option>
    </select>
    <button
      className="sc-bwzfXH iSltUj"
      onClick={[Function]}
    >
      Start
    </button>
  </div>
</div>
`;
